hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,817 @@
+
+        URL parameter:
+            - order (sale.order): The sale order model instance.
+
+        JSON response:
+            - message (str): A confirmation message indicating the action performed.
+
+        Returns:
+            dict: A dictionary with a confirmation message.
+
+        """
+        # Confirm the sale order
+        order.action_confirm()
+
+        return {"message": f"Sale order with ID: {order.id} successfully confirmed."}
+
+    @route("/create_schedule_activity", methods=["POST"], type="json", auth="user")
+    def create_schedule_activity(self):
+        """Creates a scheduled activity associated with an existing sale order.
+
+        This function creates a scheduled activity in Odoo related to a specific sale order,
+        using the details provided in the JSON request body.
+
+        JSON request body:
+            - sale_order_id (int): ID of the existing sale order.
+            - activity_type_id (int): ID of the activity type.
+            - summary (str, optional): Summary text of the activity.
+            - date_deadline (str): Deadline date for the activity in 'YYYY-MM-DD' format.
+            - note (str, optional): Note for the activity.
+            - user_id (int, optional): ID of the user assigned to the activity. If not provided, the current user's ID is used.
+
+        JSON response:
+            - message (str): A confirmation message with the ID of the scheduled activity and the sale order ID.
+
+        Returns:
+            dict: A dictionary with a confirmation message.
+        """
+        data = request.jsonrequest
+        sale_order_id = data.get("sale_order_id")  # ID of the existing sale order
+        activity_type_id = data.get("activity_type_id")  # ID of the activity type
+        summary = data.get("summary", "")  # Summary text of the activity (optional)
+        date_deadline = data.get("date_deadline")  # Deadline date for the activity
+        note = data.get("note", "")  # Note for the activity (optional)
+        user_id = data.get("user_id", request.env.uid)  # User assigned to the activity
+
+        # Create the scheduled activity
+        activity = request.env["mail.activity"].create(
+            {
+                "activity_type_id": activity_type_id,
+                "note": note,
+                "date_deadline": date_deadline,
+                "res_model_id": request.env["ir.model"]._get("sale.order").id,
+                "res_id": sale_order_id,
+                "user_id": user_id,
+                "summary": summary,
+            }
+        )
+
+        return {
+            "message": f"Scheduled activity created with ID: {activity.id} for sale order {sale_order_id}."
+        }
+
+    @route(
+        '/create_schedule_activity_invoice/<model("account.move"):invoice>',
+        methods=["POST"],
+        type="json",
+        auth="user",
+    )
+    def create_schedule_activity_invoice(self, invoice=False):
```

Comment:
same here, you can merge this method with the above, you can handle them if you do it like this
https://github.com/odoo/odoo/blob/15.0/addons/website_crm_partner_assign/controllers/main.py#L210-L214

se how they handle different records sets by model type

the thing is you can avoid duplicated code